### PR TITLE
makes provider creation fields optional and adds target info in subsc…

### PR DESCRIPTION
**Pull request by fanatic75**
…riber model

## What does this PR do?

Makes provider required field to optional when creating provider. 

Adds target identifier and providerType to subscriber response model 

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/api/messaging.php [hunk 1]
```diff
@@ -1726,8 +2020,11 @@
         $subscribers = batch(\array_map(function (Document $subscriber) use ($dbForProject) {
             return function () use ($subscriber, $dbForProject) {
                 $user = Authorization::skip(fn () => $dbForProject->getDocument('users', $subscriber->getAttribute('userId')));
+                $target = Authorization::skip(fn () => $dbForProject->getDocument('targets', $subscriber->getAttribute('targetId')));
 
                 return $subscriber
+                    ->setAttribute('identifier', $target->getAttribute('identifier'))
```
Comment (abnegate) [inline app/controllers/api/messaging.php:2026]: Let's just add the entire `target` as an attribute here since we'll need the `name` too, and that way any future attributes we need are automatically added

Comment (fanatic75) [reply app/controllers/api/messaging.php]: done this

File: .env [hunk 2]
```diff
@@ -87,7 +87,7 @@ _APP_LOGGING_PROVIDER=
 _APP_LOGGING_CONFIG=
 _APP_GRAPHQL_MAX_BATCH_SIZE=10
 _APP_GRAPHQL_MAX_COMPLEXITY=250
-_APP_GRAPHQL_MAX_DEPTH=3
+_APP_GRAPHQL_MAX_DEPTH=4
```
Comment (abnegate) [inline .env:90]: Why this change? The updated tests should only be 3 levels 🤔

**Review by abnegate — APPROVED** [review]

File: .env (hunk 2 above)
Comment (fanatic75) [reply .env:90]: There's some counting error when calculating depth in GraphQL;

**Review by abnegate — APPROVED** [review]
